src/effects/backends/builtin/glitcheffect.cpp

```diff
@@ -0,0 +1,121 @@
+    quantize->setId("quantize");
+    quantize->setName(QObject::tr("Quantize"));
+    quantize->setShortName(QObject::tr("Quantize"));
+    quantize->setDescription(QObject::tr(
+            "Round the Time parameter to the nearest 1/8 beat."));
+    quantize->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    quantize->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    quantize->setRange(0, 1, 1);
+
+    EffectManifestParameterPointer triplet = pManifest->addParameter();
+    triplet->setId("triplet");
+    triplet->setName(QObject::tr("Triplets"));
+    triplet->setShortName(QObject::tr("Triplets"));
+    triplet->setDescription(
+            QObject::tr("When the Quantize parameter is enabled, divide "
+                        "rounded 1/8 beats of Time parameter by 3."));
+    triplet->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    triplet->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    triplet->setRange(0, 0, 1);
+
+    return pManifest;
+}
+
+void GlitchEffect::loadEngineEffectParameters(
+        const QMap<QString, EngineEffectParameterPointer>& parameters) {
+    m_pDelayParameter = parameters.value("delay_time");
+    m_pQuantizeParameter = parameters.value("quantize");
+    m_pTripletParameter = parameters.value("triplet");
+}
+
+void GlitchEffect::processChannel(
+        GlitchGroupState* pGroupState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& engineParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+    // The minimum of the parameter is zero so the exact center of the knob is 1 beat.
+    double period = m_pDelayParameter->value();
+
+    int delay_frames;
+    double min_delay;
+    if (groupFeatures.has_beat_length_sec) {
+        if (m_pQuantizeParameter->toBool()) {
+            period = roundToFraction(period, 8);
+            if (m_pTripletParameter->toBool()) {
+                period /= 3.0;
+            }
+        }
+        period = std::max(period, 1 / 8.0);
+        delay_frames = static_cast<int>(period * groupFeatures.beat_length_sec *
+                engineParameters.sampleRate());
+        min_delay = 1 / 8.0 * groupFeatures.beat_length_sec * engineParameters.sampleRate();
+    } else {
+        delay_frames = static_cast<int>(period * engineParameters.sampleRate());
+        min_delay = 1 / 8.0 * engineParameters.sampleRate();
+    }
+
+    if (delay_frames < min_delay) {
+        SampleUtil::copy(
+                pOutput,
+                pInput,
+                engineParameters.samplesPerBuffer());
+        return;
+    }
+
+    int delay_samples = delay_frames * engineParameters.channelCount();
+    pGroupState->sample_count += engineParameters.samplesPerBuffer();
+    if (pGroupState->sample_count >= delay_samples) {
+        if (m_pDelayParameter->value() < 2.0) {
```

Comment:
```suggestion
        // If the delay parameter is at its maximum value, we don't update the `repeat_buf`
        // in order to achieve an "audio freeze" effect.
        if (m_pDelayParameter->value() < 2.0) {
```

Reply:
> Please define a `constexpr double maxDelay = 2.0 in the class and refer to it here`

And also here. This way we make sure this `if` stays consistent with the actual maxim value of the delay parameter (in case someone changes it in the future).